src/smc-webapp/project_files.cjsx

```diff
@@ -1536,44 +1537,76 @@ ProjectFilesActionBox = rclass
             if not (single_file_data.is_public and single_file_data.public?.path isnt single_file)
                 @share_click()
 
-    download_click: ->
+    download_single_click: ->
         @props.actions.download_file
             path : @props.checked_files.first()
             log : true
         @props.actions.set_file_action()
 
-    render_download_link: (single_item) ->
+    download_multiple_click: ->
+        destination = ReactDOM.findDOMNode(@refs.download_archive).value
+        async.series([
+            (cb) =>
+                @props.actions.zip_files
+                    src  : @props.checked_files.toArray()
+                    dest : misc.path_to_file(@props.current_path, destination)
+                    cb   : cb
+            (cb) =>
```

Comment:
This second block isn’t actually asynchronous.

Basically you want
```
@props.actions.zip_files
    cb: (err) ->
        if err 
            ....
            return
        download file
        ...
```
It’s so simple in this case that async isn’t useful.